server/src/main/java/io/crate/analyze/CopyStatementSettings.java

```diff
@@ -90,4 +120,32 @@ public static <E extends Enum<E>> E settingAsEnum(Class<E> settingsEnum, String
         }
         return Enum.valueOf(settingsEnum, settingValue.toUpperCase(Locale.ENGLISH));
     }
+
+    public static List<String> commonCopyToSettings = List.of(
+        COMPRESSION_SETTING.getKey(),
+        OUTPUT_FORMAT_SETTING.getKey(),
+        WAIT_FOR_COMPLETION_SETTING.getKey()
+    );
+
+    public static List<String> commonCopyFromSettings = List.of(
+        COMPRESSION_SETTING.getKey(),
+        INPUT_FORMAT_SETTING.getKey(),
+        WAIT_FOR_COMPLETION_SETTING.getKey(),
+        OVERWRITE_DUPLICATES_SETTING.getKey(),
+        FAIL_FAST_SETTING.getKey(),
+        SHARED_SETTING.getKey(),
+        NUM_READERS_SETTING.getKey(),
+        BULK_SIZE_SETTING.getKey(),
+        NodeFilters.NAME
+    );
+
+    // CSV specific settings.
+    // We keep them separate from common settings because
+    // we don't want '... FROM S3://... WITH (some_csv_setting)...' to pass "known setting" validation.
+    public static List<String> csvSettings = List.of(
+        EMPTY_STRING_AS_NULL.getKey(),
+        CSV_COLUMN_SEPARATOR.getKey(),
+        INPUT_HEADER_SETTINGS.getKey(),
+        CSV_COLUMN_SEPARATOR.getKey()
+    );
```

Comment:
I didn't unite those in a single place and do `Lists.concat` in planner because later on in Azure and S3 plugins we will need those granularity, it will be smth like `Lists.concat(common, azure)` and `Lists.concat(common, s3)`